posthog/management/commands/create_channel_definitions_file.py

```diff
@@ -0,0 +1,149 @@
+import json
+import re
+import subprocess
+from collections import OrderedDict
+from dataclasses import dataclass
+from enum import Enum
+from typing import Optional, Tuple
+
+from django.core.management.base import BaseCommand
+
+
+OUTPUT_FILE = "posthog/models/channel_type/channel_definitions.json"
+
+
+class EntryKind(str, Enum):
+    source = "source"
+    medium = "medium"
+
+
+@dataclass()
+class SourceEntry:
+    hostname_type: Optional[str]
+    type_if_paid: Optional[str]
+    type_if_organic: Optional[str]
+
+
+class Command(BaseCommand):
+    help = (
+        "Write the channel_definitions.json file. Needs a ga4 sources file like"
+        "https://storage.googleapis.com/support-kms-prod/qn1xhBu8MVcZPIZ2WZMNdI40FtZXFPGYxj2K"
+        "as input. The best way I have found to do this is to open it in Google Docs, then save it as a text file"
+    )
```

Comment:
How often should this `channel_definitions.json` update be done? Thinking about long-term maintainability of this mechanism

Reply:
About once a year according to google, though we might want to update more frequently than that if we get requests from customers to add or change something

Reply:
I wrote this in the comment :)